part_one.sql

```diff
@@ -1 +1,93 @@
 
+/* Prompt 1 */
+
+SELECT
+   document.id AS document_id 
+FROM
+   document 
+   LEFT JOIN
+      (
+         SELECT DISTINCT
+            document_id 
+         FROM
+            `page`
+      )
+      AS pages 
+      ON pages.document_id = document.id 
+WHERE
+   pages.document_id IS NULL;
+
+/* Prompt 2 */
+
+SELECT
+   report.title AS report_title,
+   B.total_count AS page_count
+FROM
+   report 
+   LEFT JOIN
+      (
+         SELECT
+            report_id,
+            SUM(A.page_count) AS total_count 
+         FROM
+            document
+            LEFT JOIN
+               (
+                  SELECT
+                     document_id,
+                     COUNT(*) AS page_count
+                  FROM
+                     `page` 
+                  GROUP BY
+                     document_id
+               )
+               AS A 
+               ON A.document_id = document.id 
+         WHERE
+            report_id 
+         GROUP BY
+            report_id
+      )
+      AS B 
+      ON B.report_id = report.id 
+WHERE
+   B.total_count IS NOT NULL
+ORDER BY
+	report.title;
```

Comment:
`LEFT JOINS` working from the inside out made the most sense to me; however, if some of these tables were significantly larger than the others, I would consider using a subquery with `something_id IN` rather than a join. Ideally would have wanted to figure out a way to filter out the null `B.total_count` before the last step, but couldnt figure that out while keeping the `GROUP BY` aggregation